src/Compilers/CSharp/Portable/Symbols/PublicModel/TypeSymbol.cs

```diff
@@ -111,7 +111,7 @@ ISymbol ITypeSymbol.FindImplementationForInterfaceMember(ISymbol interfaceMember
                 : null;
         }
 
-        bool ITypeSymbol.IsUnmanagedType => !UnderlyingTypeSymbol.IsManagedType;
+        bool ITypeSymbol.IsUnmanagedType => !UnderlyingTypeSymbol.IsManagedTypeNoUseSiteDiagnostics;
```

Comment:
Seems like there is a gap now between what we have interntally and what we expose through the public API. Is there a need to give this info to our API consumers?

Reply:
> Seems like there is a gap now between what we have interntally and what we expose through the public API. Is there a need to give this info to our API consumers?

We have a bunch of API like that. We do not expose use-site diagnostics or any other diagnostics for any of them. They do not return diagnostics. APIs that return diagnostics will incorporate use-site diagnostics naturally.

---
In reply to: [460266634](https://github.com/dotnet/roslyn/pull/46114#discussion_r460266634) [](ancestors = 460266634)